services/venice-controller/src/main/java/com/linkedin/venice/controller/kafka/TopicCleanupService.java

```diff
@@ -135,10 +122,26 @@ public TopicCleanupService(
     } else {
       this.sourceOfTruthPubSubAdminAdapter = null;
     }
+
+    if (!this.admin.isParent()) {
```

Comment:
is keyword `this` here necessary? My understanding for our code convention is that `this` is used when there is ambiguity or cases it has to be used.

Reply:
What's main reason to move the child controller api calls into constructor?

Reply:
the reason is that the value of `localDatacenter` does not change so we do not need to re-calculate it everytime topics are deleted. cc @xunyin8 correct me if I missed anything